src/memory.jl

```diff
@@ -327,6 +327,70 @@ for (f, srcPtrTy, dstPtrTy) in (("cuMemcpyDtoH", CuPtr, Ptr),
     end
 end
 
+function copy_3d(src::Union{Ptr,CuPtr}, srcTyp::Type{<:Buffer},
+                 dst::Union{Ptr,CuPtr}, dstTyp::Type{<:Buffer},
+                 width::Integer, height::Integer=1, depth::Integer=1;
```

Comment:
`width` is in bytes while `height` and dept `depth` are just number of elements, as in the draft in the corresponding issue #149, right? If so, I think it is clearer to call it `widthInBytes` as there. If it is called just `width`, then I would expect that it is a number of elements, like `height` and dept `depth`, which is then automatically multiplied with the size of an element when passed further to the underlying CUDA call. 
Even if this `widthInBytes` is maybe not the nicest thing to have, I believe it is better if the arguments have the same meaning as in the corresponding [CUDA C function](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g79f4f3fde6ae0f529568d881d9e11987). This will probably prevent some bugs when people translate from CUDA C to Julia using CUDAdrv...

Reply:
I was mimicking the CUDA runtime API, which is supposedly what people will be porting from: http://developer.download.nvidia.com/compute/cuda/2_3/toolkit/docs/online/group__CUDART__MEMORY_gc1372614eb614f4689fbb82b4692d30a.html
There, the elements (when using pointers) are `unsigned char` aka bytes, so it seemed reasonable to just call it `width`?

Reply:
OK, I understand. It is clearly written in the [cudaExtent Struct Reference](http://developer.download.nvidia.com/compute/cuda/2_3/toolkit/docs/online/structcudaExtent.html) that all `width`, `height` and `depth` are supposed to be given in bytes. However, from my tests and experience, I conclude that this is an error / bad wording in the documentation. I think that `height` and `depth` have to be just the number of elements, not the amount of bytes (i.e. as in the corresponding CUDA C function). In the end, it does not make that much sense to give the second or third dimension in bytes, does it?

If I change for example:
```
Mem.copy_3d(pointer(B), Mem.Device, pointer(A), Mem.Host, nx*sizeof(A[1]), 1, 
nz,...
```
to
```
Mem.copy_3d(pointer(B), Mem.Device, pointer(A), Mem.Host, nx*sizeof(A[1]), 1, 
nz*sizeof(A[1]),...
```
it fails:
```
jl_CUDAdrv6> jul -O3 --check-bounds=no h_d_transfers_3d_cpy3d_allbytes.jl
ERROR: LoadError: CUDA error: invalid argument (code 1, ERROR_INVALID_VALUE)
Stacktrace:
 [1] throw_api_error(::CUDAdrv.cudaError_enum) at /users/omlins/.julia/1.4.1/dom-gpu/dev/CUDAdrv/src/error.jl:106
 [2] macro expansion at /users/omlins/.julia/1.4.1/dom-gpu/dev/CUDAdrv/src/error.jl:113 [inlined]
 [3] cuMemcpy3DAsync_v2(::Base.RefValue{CUDAdrv.CUDA_MEMCPY3D_st}, ::CuStream) at /scratch/snx3000tds/omlins/9_soft_install_julia_1_4/software/Julia/1.4.1-CrayGNU-19.10-cuda-10.1/extensions/packages/CUDAapi/XuSHC/src/call.jl:93
 [4] #copy_3d#9 at /users/omlins/.julia/1.4.1/dom-gpu/dev/CUDAdrv/src/memory.jl:386 [inlined]
 [5] top-level scope at /users/omlins/0_misc/hpc_julia/0_h_d_transfers/jl_CUDAdrv6/h_d_transfers_3d_cpy3d_allbytes.jl:22
 [6] include(::Module, ::String) at ./Base.jl:377
 [7] exec_options(::Base.JLOptions) at ./client.jl:288
 [8] _start() at ./client.jl:484
```
Also, we are using `CUDAdrv.@apicall(:cuMemcpy3D, ...)` since a long time with correct results passing only `width` in bytes and `height` and `depth` as number of elements. This calls the function from the CUDA runtime API, doesn't it?

So, if you agree that is an error / bad wording in the CUDA runtime API documentation, then I would use `widthInBytes` to help limiting the confusion that might appear due to this part of the CUDA runtime API documentation